From 78947d73f55a83f023acd653e04f4cb28c1c6ddc Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Tue, 12 Oct 2021 10:41:58 +0200
Subject: [PATCH 117/120] pinctrl: sunxi: sunxi_pinctrl_irq_ack avoid build
 warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Avoid a build warning

drivers/pinctrl/sunxi/pinctrl-sunxi.c:
In function ‘sunxi_pinctrl_irq_ack’:
drivers/pinctrl/sunxi/pinctrl-sunxi.c:1035:2: warning:
ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
 1035 |  u32 new = readl(pctl->membase + status_reg);
      |  ^~~

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 drivers/pinctrl/sunxi/pinctrl-sunxi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/sunxi/pinctrl-sunxi.c b/drivers/pinctrl/sunxi/pinctrl-sunxi.c
index 1a58dcfc1b9a..0c1aca79a624 100644
--- a/drivers/pinctrl/sunxi/pinctrl-sunxi.c
+++ b/drivers/pinctrl/sunxi/pinctrl-sunxi.c
@@ -1026,13 +1026,14 @@ static void sunxi_pinctrl_irq_ack(struct irq_data *d)
 	struct sunxi_pinctrl *pctl = irq_data_get_irq_chip_data(d);
 	u32 status_reg = sunxi_irq_status_reg(pctl->desc, d->hwirq);
 	u8 status_idx = sunxi_irq_status_offset(d->hwirq);
+	u32 new, old;
 
-	u32 old = readl(pctl->membase + status_reg);
+	old = readl(pctl->membase + status_reg);
 
 	/* Clear the IRQ */
 	writel(1 << status_idx, pctl->membase + status_reg);
 
-	u32 new = readl(pctl->membase + status_reg);
+	new = readl(pctl->membase + status_reg);
 
 	pr_err("acked %ld in 0x%08x, was 0x%08x, now 0x%08x\n",
 		d->hwirq, status_reg, old, new);
-- 
2.32.0

